core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -304,6 +329,42 @@ void processSubchannelState(SubchannelData subchannelData, ConnectivityStateInfo
     }
   }
 
+  /**
+   * Only called after all addresses attempted and failed (TRANSIENT_FAILURE).
+   */
+  private void scheduleBackoff() {
+    if (!serializingRetries) {
+      return;
+    }
+
+    class EndOfCurrentBackoff implements Runnable {
+      @Override
+      public void run() {
+        reconnectTask = null;
+        if (rawConnectivityState == SHUTDOWN) {
+          return;
+        }
+        addressIndex.reset();
+        requestConnection();
+      }
+    }
+
+    // Just allow the previous one to trigger when ready if we're already in backoff
+    if (reconnectTask != null) {
+      return;
+    }
+
+    if (reconnectPolicy == null) {
+      reconnectPolicy = bkoffPolProvider.get();
```

Comment:
I see the field is initialized here, but it isn't ever cleared on success. InternalSubchannel does that in cancelReconnectTask().